cmd/oras/internal/option/spec.go

```diff
@@ -33,22 +36,38 @@ type ImageSpec struct {
 	PackVersion oras.PackManifestVersion
 }
 
-// Parse parses flags into the option.
-func (opts *ImageSpec) Parse() error {
-	switch opts.flag {
+// Set validates and sets the flag value from a string argument.
+func (i *ImageSpec) Set(value string) error {
+	i.flag = value
+	switch value {
 	case ImageSpecV1_1:
-		opts.PackVersion = oras.PackManifestVersion1_1_RC4
+		i.PackVersion = oras.PackManifestVersion1_1_RC4
 	case ImageSpecV1_0:
-		opts.PackVersion = oras.PackManifestVersion1_0
+		i.PackVersion = oras.PackManifestVersion1_0
 	default:
-		return fmt.Errorf("unknown image specification flag: %q", opts.flag)
+		return &oerrors.Error{
+			Err:            fmt.Errorf("unknown image specification flag: %s", value),
+			Recommendation: fmt.Sprintf("Available options: %s", i.Type()),
+		}
 	}
 	return nil
 }
 
+// Type returns the type of the flag.
+func (i *ImageSpec) Type() string {
+	return fmt.Sprintf("%s, %s", ImageSpecV1_1, ImageSpecV1_0)
+}
```

Comment:
The `Type()` should be `string`. Otherwise, it is hacking and you need to document a lot.